spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/AbstractDeployer.java

```diff
@@ -92,12 +97,14 @@ public D save(D definition) {
 		String name = definition.getName();
 		String def = definition.getDefinition();
 		validateBeforeSave(name, def);
-		List<ModuleDescriptor> moduleDescriptors = parser.parse(name, def, definitionKind);
-
-		// todo: the result of parse() should already have correct (polymorphic) definitions
-		List<ModuleDefinition> moduleDefinitions = createModuleDefinitions(moduleDescriptors);
-		if (!moduleDefinitions.isEmpty()) {
-			definition.setModuleDefinitions(moduleDefinitions);
+		if(!ComposedJobUtil.isComposedJobDefinition(def)){
+			List<ModuleDescriptor> moduleDescriptors = moduleDescriptors = 
```

Comment:
but why are you setting a variable equal to itself is the part I'm not understanding...